client/testing.go

```diff
@@ -0,0 +1,70 @@
+package client
+
+import (
+	"testing"
+
+	"github.com/cloudquery/cq-provider-sdk/logging"
+	"github.com/cloudquery/cq-provider-sdk/provider"
+	"github.com/cloudquery/cq-provider-sdk/provider/schema"
+	providertest "github.com/cloudquery/cq-provider-sdk/provider/testing"
+	"github.com/golang/mock/gomock"
+	"github.com/hashicorp/go-hclog"
+)
+
+func K8sTestHelper(t *testing.T, table *schema.Table, snapshotDirPath string) {
+	cfg := ``
+	providertest.TestResource(t, providertest.ResourceTestCase{
+		Provider: &provider.Provider{
+			Name:      "k8s_mock_test_provider",
+			Version:   "development",
+			Configure: Configure,
+			Config: func() provider.Config {
+				return &Config{}
+			},
+			ResourceMap: map[string]*schema.Table{
+				"test_resource": table,
+			},
+		},
+		Table:           table,
+		Config:          cfg,
+		SkipEmptyRows:   false,
+		SkipEmptyColumn: true,
+	})
+}
+
+type TestOptions struct {
+	SkipEmptyJsonB bool
+}
+
+func K8sMockTestHelper(t *testing.T, table *schema.Table, builder func(*testing.T, *gomock.Controller) Services, options TestOptions) {
+	t.Helper()
+	ctrl := gomock.NewController(t)
+
+	cfg := ``
+
+	providertest.TestResource(t, providertest.ResourceTestCase{
+		Provider: &provider.Provider{
+			Name:    "aws_mock_test_provider",
+			Version: "development",
+			Configure: func(logger hclog.Logger, _ interface{}) (schema.ClientMeta, error) {
+				c := &Client{
+					Log: logging.New(&hclog.LoggerOptions{
+						Level: hclog.Warn,
```

Comment:
why not reuse the same logger here? I think in test the `providertest.TestResource` will call `Configure` with [`testlogger`](https://github.com/cloudquery/cq-provider-sdk/tree/main/testlog) which will eventually call to `t.Log` and will be useful as go automatically knows to print logs only for tests that fail.

Reply:
fixed